source/characterProcessing.py

```diff
@@ -519,6 +519,35 @@ def __init__(self, locale):
 			log.error("Invalid complex symbol regular expression in locale %s: %s" % (locale, e))
 			raise LookupError
 
+	def _replaceGroups(self, m: re.Match, string: str) -> str:
+		"""Replace matching group references (\\1, \\2, ...) with the corresponding matched groups.
+		Also replace \\\\ with \\ and reject other escapes, for escaping coherency.

```

Comment:
Is there a test for the `\\\\` case?

Reply:
I added one indeed, better check that quadruple backslash works.
Note that here in the doc string, `\\\\` is actually only two characters.